core/src/main/java/org/vertexium/util/GeoUtils.java

```diff
@@ -0,0 +1,144 @@
+package org.vertexium.util;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.locationtech.jts.algorithm.Orientation;
+import org.locationtech.jts.geom.*;
+import org.locationtech.jts.operation.valid.IsValidOp;
+import org.locationtech.jts.operation.valid.TopologyValidationError;
+import org.vertexium.VertexiumException;
+import org.vertexium.type.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GeoUtils {
+    private static final VertexiumLogger LOGGER = VertexiumLoggerFactory.getLogger(CloseableUtils.class);
+    private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory();
+
+    public static boolean intersects(GeoShape left, GeoShape right) {
+        return toJtsGeometry(left).intersects(toJtsGeometry(right));
+    }
+
+    public static boolean within(GeoShape left, GeoShape right) {
+        return toJtsGeometry(left).within(toJtsGeometry(right));
+    }
+
+    public static GeoShape toGeoShape(Geometry geometry, String description) {
+        if (geometry instanceof GeometryCollection) {
+            GeoCollection geoCollection = new GeoCollection(description);
+            GeometryCollection geometryCollection = (GeometryCollection) geometry;
+            for (int i = 0; i < geometryCollection.getNumGeometries(); i++) {
+                geoCollection.addShape(toGeoShape(geometryCollection.getGeometryN(i), null));
+            }
+            return geoCollection;
+        }
+
+        if (geometry instanceof Polygon) {
```

Comment:
Either make this an `else if` or change the `else if` below to `if`.  I personally like just using `if` since there are return statements inside.

Reply:
Good call. Fixed in https://github.com/visallo/vertexium/pull/445/commits/22792556a563f2af4d2573cc86a1469ee575b927